src/view/components/AddRestaurantForm.ts

```diff
@@ -0,0 +1,159 @@
+import generateSelectBox from "../generateComponent/generateSelectBox";
+import generateButton from "../generateComponent/generateButton";
+import FormItem from "./FormItem";
+import createElementByTag from "../generateComponent/utils/createElementByTag";
+import RestaurantList from "../../domain/RestaurantList";
+import { CATECORY, DISTANCE } from "../../constants/selectOptions";
+import FORM_ITEM_TEXTS from "../../constants/formItemTexts";
+class AddRestaurantForm {
+  #categoryFormItem: FormItem;
+  #nameFormItem: FormItem;
+  #distanceFormItem: FormItem;
+  #descriptionFormItem: FormItem;
+  #urlFormItem: FormItem;
+
+  constructor() {
+    this.#categoryFormItem = this.#createcategoryFormItem();
+    this.#nameFormItem = this.#createNameFormItem();
+    this.#distanceFormItem = this.#createDistanceFormItem();
+    this.#descriptionFormItem = this.#createDescriptionFormItem();
+    this.#urlFormItem = this.#createUrlFormItem();
+  }
+
+  getForm({ hideModalFunc }: { hideModalFunc: () => void }) {
+    const form = document.createElement("form");
+    form.append(
+      this.#categoryFormItem.item,
+      this.#nameFormItem.item,
+      this.#distanceFormItem.item,
+      this.#descriptionFormItem.item,
+      this.#urlFormItem.item,
+      this.#createButtonDiv(hideModalFunc)
+    );
+
+    return form;
+  }
+
+  resetForms() {
+    this.#categoryFormItem.resetForm();
+    this.#nameFormItem.resetForm();
+    this.#distanceFormItem.resetForm();
+    this.#descriptionFormItem.resetForm();
+    this.#urlFormItem.resetForm();
+  }
+
+  #createcategoryFormItem() {
+    const categorySelectBoxInModal = generateSelectBox(CATECORY, false);
+    const categoryFormItem = new FormItem({
+      subject: FORM_ITEM_TEXTS.categoryTitle,
+      readableElement: categorySelectBoxInModal,
+      isRequired: true,
+    });
+
+    return categoryFormItem;
+  }
+
+  #createNameFormItem() {
+    const nameInput = document.createElement("input");
+    nameInput.type = "text";
+    nameInput.name = "name";
+    nameInput.id = "name";
+    nameInput.required = true;
+
+    const nameFormItem = new FormItem({
+      subject: FORM_ITEM_TEXTS.nameTitle,
+      readableElement: nameInput,
+      isRequired: true,
+    });
+    return nameFormItem;
+  }
+
+  #createDistanceFormItem() {
+    const DISTANCE_STRINGS = DISTANCE.map((el) => el.toString());
```

Comment:
이런식으로 축약해서도 사용가능하니 참고하시면 좋겠네용
```suggestion
    const DISTANCE_STRINGS = DISTANCE.map(String);
```